modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C

```diff
@@ -831,6 +838,32 @@ PatternedCartesianMeshGenerator::generate()
         out_mesh->add_point(p_tmp, node_azi_list[i * side_intervals + j - 1].second);
       }
     }
+
+    // adjust quadratic mid-edge nodes after boundary node redistribution
+    if (_boundary_quad_elem_type != QUAD_ELEM_TYPE::QUAD4)
+    {
+      for (auto elem = out_mesh->elements_begin(); elem != out_mesh->elements_end(); ++elem)
+      {
+        // only adjust elements on outer boundary
+        if (!(*elem)->on_boundary())
+          continue;
+
+        // adjust point 5
+        Point pt_5 = ((*elem)->point(1) + (*elem)->point(2)) / 2.0;
+        out_mesh->add_point(pt_5, (*elem)->node_ptr(5)->id());
+
+        // adjoin point 7
+        Point pt_7 = ((*elem)->point(0) + (*elem)->point(3)) / 2.0;
```

Comment:
why are we moving them?

Reply:
The existing algorithm generates entire boundary region, and then if the `uniform_mesh_on_sides` option is used then the MG redistributes the boundary nodes on each side so that they are uniformly spaced. When the elements are QUAD4 this is fine, but for QUAD8/9 the mid-edge nodes staying in their original locations results in weird looking elements. This section moves the mid-edge nodes to midpoints based on the new boundary node locations so the element shape is better.

Reply:
let's have some of this explanation as a comment right before we do it in both MGs?

Reply:
added